gCoffee/app/src/main/java/com/mborowiec/gcoffee/Order.java

```diff
@@ -150,6 +150,46 @@ private void handleDeepLinks(Uri data) {
                     Toast.LENGTH_SHORT).show();
             }
     }
+
+    /**
+     * Normalizes the string by removing punctuation, removing whitespaces
+     * and making the string lowercase.
+     * @param input input string
+     * @return a normalized string
+     */
+    public String normalize(String input) {
+        String normalized = input.replaceAll("[^a-zA-Z ]" , "")
+                .replaceAll("\\s+", " ")
+                .toLowerCase();
+
+        return normalized;
+    }
+
+    /**
+     * Calculates the Levenshtein distance between two strings
+     * @param str1 first input string
+     * @param str2 second input string
+     * @return distance between strings
+     */
+    public Integer distance(String str1, String str2) {
+        //TODO: calculate Levenshtein Distance between two strings
+
+        return null;
+    }
+
+    /**
+     * Looks for the closest match (with closest distance) in the menu for an order with typos.
```

Comment:
The comment let's me assume that it only works for orders with typos, but I assume that those without typos should work as well?